### PR TITLE
Update go waku

### PR DESCRIPTION
**Pull request by cammellos**
Updates go waku to avoid a crash, it's using an off brach version as per latest version some tests in waku2 are failing, and we are close to release cutting and we want to avoid extra changes.

The changes include recovering from a panic (bad but the other solution proposed was locking in tests), and soft fail on stop when unsubscribing from peers, since it currently was trying to wait for the peers to acknowledge, but that's no necessary when logging out and increase logout time by up to 30s.

### CODE REVIEW

File: protocol/messenger.go [hunk 1]
```diff
@@ -521,7 +521,17 @@ func NewMessenger(
 			pushNotificationClient.Stop,
 			communitiesManager.Stop,
 			encryptionProtocol.Stop,
-			transp.ResetFilters,
+			func() error {
+				ctx, cancel := context.WithTimeout(context.Background(), time.Second*1)
```
Comment (Samyoul) [inline protocol/messenger.go]: Nothing major, but you don't need to multiply by 1 🤷 

```suggestion
				ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

**Review by Samyoul — APPROVED** [review]
That was a lot of CPU and memory for basically 1 function worth of review 🤣

File: protocol/messenger.go (hunk 1 above)
Comment (cammellos) [reply protocol/messenger.go]: ouch :)

**Comment by qfrank** [comment]
any chance to write a test to reproduce the issue and prove it get solved? @cammellos

**Review by qfrank — APPROVED** [review]

**Comment by cammellos** [comment]
@qfrank it was caught by our mobile integration tests, so it's partially cover there, I wasn't quite able to replicate it here, but I can try again, though not on this PR as this is a release blocker and would like to get it in as soon as possible

**Comment by qfrank** [comment]
in my memory, it was the second time we got the like issue 🥹

**Comment by cammellos** [comment]
> in my memory, it was the second time we got the like issue 🥹

different issue, albeit similar, both times was caught by our integration tests, the previous one was sending on a closed channel, this one was closing a closed channel.
The previous one was addressed by making sure the owner of a channel is also the reader (it should always be like that), this one is due to some concurrency where start/stop/start/stop are interleaved I believe, but could not replicate yet

**Comment by qfrank** [comment]
It's highly possible that we have different memories 😉. I'm referring to this: https://github.com/waku-org/go-waku/pull/553. The similarity is that there's a delay when logging out.

**Comment by cammellos** [comment]
oh @qfrank the dead loop is not in master, it's still a PR that we were trying to avoid the crash

**Pull request merged**